org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/lookup/BinaryTypeBinding.java

```diff
@@ -87,26 +88,31 @@
 */
 
 @SuppressWarnings({ "rawtypes", "unchecked" })
-public class BinaryTypeBinding extends ReferenceBinding {
+//AspectJ Extension - XXX extending SourceTypeBinding is a HORRIBLE hack, was 'extends ReferenceBinding'
+public class BinaryTypeBinding extends SourceTypeBinding {
 
 	public static final char[] TYPE_QUALIFIER_DEFAULT = "TypeQualifierDefault".toCharArray(); //$NON-NLS-1$
 
 	private static final IBinaryMethod[] NO_BINARY_METHODS = new IBinaryMethod[0];
 
 	// all of these fields are ONLY guaranteed to be initialized if accessed using their public accessor method
-	protected ReferenceBinding superclass;
+// AspectJ Extension - comment out some fields
+//	protected ReferenceBinding superclass;
 	protected ReferenceBinding enclosingType;
-	protected ReferenceBinding[] superInterfaces;
+//	protected ReferenceBinding[] superInterfaces;
+	// AspectJ - TODO kriegaex: Is there anything to be done if this new field is active?
```

Comment:
Actually, I do not understand your comment. Would I be asking too much for you to just add a commit to this branch, fixing it the way you deem correct? It would automatically become part of the PR. I permitted you to collaborate here.

Reply:
OK, looking at the code in the IDE, I understand a bit better (but not fully) what you mean. But I checked in class `SourceTypeBinding`, and there is no `ReferenceBinding[] permittedSubtypes`, but a pretty recent `ReferenceBinding[] permittedTypes`, which sounds quite similar. There also is a corresponding setter/getter pair for that field.

Further down, in this file here I also see `private BinaryTypeBinding prototype`, which has existed since 2013. In the parent class I see `protected SourceTypeBinding prototype`, about a month older, also associated with a regular getter method and a boolean `isPrototype()` method. It does not look like we ever did anything there, even though the property name is the same (the type is slightly different, though). I really have no idea what this stuff does, I need you to decide what to do here.

Reply:
> I need you to decide what to do here.

I have no idea, but I don't see it as a preventing the commit merge, as you tell me all downstream tests are passing. I was just trying to explain what I *think* I recall around these changes. I doubt we have sufficient tests around binary sources, particularly for anything recently added to Java. Sounds like prototype is also being mishandled.